fj-host/src/platform.rs

```diff
@@ -0,0 +1,45 @@
+// Represents platform trait
+trait Platform {
+    fn file_name(name: &str) -> String;
+}
+
+// Represents all platforms supported
+
+// Mac OS
+struct Macos;
+// Windows
+struct Windows;
+// Linux
+struct Unix;
+
+impl Platform for Windows {
+    fn file_name(name: &str) -> String {
+        format!("{}.dll", name)
+    }
+}
+impl Platform for Macos {
+    fn file_name(name: &str) -> String {
+        format!("lib{}.dylib", name)
+    }
+}
+impl Platform for Unix {
+    fn file_name(name: &str) -> String {
+        format!("lib{}.so", name)
+    }
+}
+
+// Represents common apis availiable independent of hosts
+pub struct HostPlatform;
+
+impl HostPlatform {
+    pub fn host_file_name(name: &str) -> String {
+        if cfg!(windows) {
+            Windows::file_name(name)
+        } else if cfg!(target_os = "macos") {
+            Macos::file_name(name)
+        } else {
+            //Unix
+            Unix::file_name(name)
+        }
+    }
+}
```

Comment:
This is fine, as long as `Platform` only has one method, but if there were more, we'd have to repeat this `if` expression in all of them. How about something like this:
``` rust
pub fn platform() -> impl Platform {
    if cfg!(windows) {
        Windows
    } else if cfg!(target_os = "macos") {
        Macos
    } else {
        Unix
    }
}
```

That would require the trait methods in `Platform` to have a `&self` parameter, so they can be called on the return value, but I think this would be fine.